mp/src/game/shared/baseplayer_shared.cpp

```diff
@@ -869,6 +871,64 @@ void CBasePlayer::Weapon_SetLast( CBaseCombatWeapon *pWeapon )
 	m_hLastWeapon = pWeapon;
 }
 
+#ifdef NEO
+void CBasePlayer::UpdateMuzzleFlashProperties(CBaseCombatWeapon* pWeapon)
+{
+	auto neoWep = static_cast<CNEOBaseCombatWeapon*>(pWeapon);
+	if (!neoWep)
+		return;
+
+	CNEOPredictedViewModelMuzzleFlash* neoViewModelMuzzleflash = static_cast<CNEOPredictedViewModelMuzzleFlash*>(m_hViewModel[MUZZLE_FLASH_VIEW_MODEL_INDEX].Get());
+	if (!neoViewModelMuzzleflash)
+		return;
+	
+	if (neoWep->GetNeoWepBits() & (NEO_WEP_DETPACK | NEO_WEP_GHOST | NEO_WEP_FRAG_GRENADE | NEO_WEP_KNIFE | NEO_WEP_SMOKE_GRENADE | NEO_WEP_SUPPRESSED | NEO_WEP_EXPLOSIVE))
```

Comment:
we should probably have a NEO_WEP_FIREARM, but ill just do this for now

Reply:
Thanks. Could you also look at caching the `GetNeoWepBits()` into a variable before the if-elif block, and then using that variable instead of calling it again for each case. Currently, it generates 5 virtual calls to the same function, when we could just store the value once and avoid the 4 extra calls:

```asm
                             ; neoWep->GetNeoWepBits() called here
        0024381f ff 90 b8        CALL       dword ptr [EAX + 0x6b8]
                 06 00 00


                             ; (skip...)


                             ; neoWep->GetNeoWepBits() called here
        00243846 ff 90 b8        CALL       dword ptr [EAX + 0x6b8]

                 06 00 00
        0024384c 83 c4 10        ADD        ESP,0x10
        0024384f a9 80 00        TEST       EAX,0x1020080
                 02 01
        00243854 0f 85 d6        JNZ        LAB_00243930
                 00 00 00
        0024385a 83 ec 0c        SUB        ESP,0xc
        0024385d 8b 03           MOV        EAX,dword ptr [EBX]
        0024385f 53              PUSH       EBX

                             ; neoWep->GetNeoWepBits()  called here
        00243860 ff 90 b8        CALL       dword ptr [EAX + 0x6b8]
                 06 00 00
        00243866 83 c4 10        ADD        ESP,0x10
        00243869 a9 00 00        TEST       EAX,BYTE_ARRAY_007ffee9[279]
                 80 00
        0024386e 75 78           JNZ        LAB_002438e8
        00243870 83 ec 0c        SUB        ESP,0xc
        00243873 8b 03           MOV        EAX,dword ptr [EBX]
        00243875 53              PUSH       EBX

                             ; neoWep->GetNeoWepBits()  called here
        00243876 ff 90 b8        CALL       dword ptr [EAX + 0x6b8]
                 06 00 00
        0024387c 83 c4 10        ADD        ESP,0x10
        0024387f a9 10 00        TEST       EAX,__DT_REL[121666].r_offset
                 10 00
        00243884 0f 85 e6        JNZ        LAB_00243970
                 00 00 00
        0024388a 83 ec 0c        SUB        ESP,0xc
        0024388d 8b 03           MOV        EAX,dword ptr [EBX]
        0024388f 53              PUSH       EBX

                             ; neoWep->GetNeoWepBits()  called here
        00243890 ff 90 b8        CALL       dword ptr [EAX + 0x6b8]
```